cmd/clusterctl/pkg/client/cluster/components.go

```diff
@@ -21,15 +21,21 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/klog"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/repository"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // ComponentsClient has methods to work with provider components in the cluster.
 type ComponentsClient interface {
```

Comment:
Yes, `templatesClient` and `templates` switched to singular because 1 YAML --> 1 template, but in this case 1 YAML --> N components so IMO we should use the plural.
But no problem to change if you prefer singular also here.